lib/ast.js

```diff
@@ -91,7 +91,15 @@ var AST_Token = DEFNODE("Token", "type value line col pos endline endcol endpos
 }, null);
 
 var AST_Node = DEFNODE("Node", "start end", {
-    clone: function() {
+    clone: function(deep) {
+        if (deep) {
+            var self = this.clone();
+            return self.transform(new TreeTransformer(function(node) {
+                if (node !== self) {
+                    return node.clone(true);
```

Comment:
@alexlamsl Just an FYI, this will only clone the `props` passed to `DEFNODE()` for each AST node class, and will only shallow clone those props. A proper deep clone is rather tricky.